src/basic/io-util.h

```diff
@@ -33,22 +33,27 @@ static inline size_t IOVEC_TOTAL_SIZE(const struct iovec *i, size_t n) {
         return r;
 }
 
-static inline size_t IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
-        size_t j;
+static inline bool IOVEC_INCREMENT(struct iovec *i, size_t n, size_t k) {
+        /* Returns true if there is nothing else to send (bytes written cover all of the iovec),
+         * false if there's still work to do. */
```

Comment:
hmm, EOF is usually indicated by commands such as read() via a ret val of 0, i.e. false. i.e. wouldn't it make more sense to invert the return value here? i.e. return true if there's more, and false if there is not?

Reply:
I think the analogy with read() is pretty weak: with read() we return how many bytes were available, and interpret 0 to mean that none were so we're done. Here, we are reporting the result of incrementing: did we manage to increment all the way to the end (return true), or did we fall short (return false). I can invert it if you still prefer, but I think the current approach makes sense.

Reply:
I merged this without changing this. If you still think it should be changed, let's do that separately.